src/main/scala/cromwell/util/GcsConnector.scala

```diff
@@ -0,0 +1,112 @@
+package cromwell.util
+
+import java.io._
+
+import com.google.api.client.auth.oauth2.Credential
+import com.google.api.client.extensions.java6.auth.oauth2.{AuthorizationCodeInstalledApp, VerificationCodeReceiver}
+import com.google.api.client.googleapis.auth.oauth2.{GoogleAuthorizationCodeFlow, GoogleClientSecrets}
+import com.google.api.client.googleapis.extensions.java6.auth.oauth2.GooglePromptReceiver
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.http.{InputStreamContent, HttpTransport}
+import com.google.api.client.json.JsonFactory
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.client.util.store.FileDataStoreFactory
+import com.google.api.services.storage.model.Bucket
+import com.google.api.services.storage.{Storage, StorageScopes}
+
+/**
+ * Provides a connector which can authenticate with Google and provide access to Google Cloud Storage
+ *
+ * @param applicationName The application name to run under
+ * @param secretFile The path to the secret file - must be local and must be JSON
+ */
+class GcsConnector(applicationName: String, secretFile: String) {
+
+  // Create the data store directory:
+  private final val DATA_STORE_DIR: File = new File(System.getProperty("user.home"), ".store/storage_sample")
+
+  // Create the JSON factory:
+  private final val JSON_FACTORY: JsonFactory = JacksonFactory.getDefaultInstance
```

Comment:
In this case I realize the reason for usage is "I'm learning how to scala". I just didn't want to say "it gives no benefit" in case I wasn't fully correct and you swooped in to show how wrong I was :)

My understanding of `final` in scala is that it can be used to a) resolve the goofy inlining hacks that java people had to deal with and b) prevent inheritance. Neither seems to be the case here, in the latter situation these are private members anyways.